cardinal/server/server_test.go

```diff
@@ -99,6 +99,84 @@ func makeTestTransactionHandler(t *testing.T, world *ecs.World, opts ...Option)
 	}
 }
 
+func TestShutDownViaMethod(t *testing.T) {
+	go func() {
+		select {
+		case <-time.After(10 * time.Second):
+			panic("took too long to shutdown.")
+		}
+	}() // If this test is frozen then it failed to shut down, create failure with panic.
+	w := inmem.NewECSWorldForTest(t)
+	sendTx := ecs.NewTransactionType[SendEnergyTx, SendEnergyTxResult]("sendTx")
+	assert.NilError(t, w.RegisterTransactions(sendTx))
+	w.AddSystem(func(world *ecs.World, queue *ecs.TransactionQueue, _ *ecs.Logger) error {
+		return nil
+	})
+	assert.NilError(t, w.LoadGameState())
+	txh := makeTestTransactionHandler(t, w, DisableSignatureVerification())
+	for !txh.IsServerRunning() {
+		//wait until server is running
+		time.Sleep(1 * time.Millisecond)
+	}
+	ctx := context.Background()
+	w.StartGameLoop(ctx, 1*time.Second)
+	for !w.IsGameLoopRunning() {
+		//wait until game loop is running.
+		time.Sleep(1 * time.Millisecond)
+	}
+	shutdownObject := NewShutdownManager(w, txh.Handler)
+	err := shutdownObject.Shutdown()
+	assert.NilError(t, err)
+	//wait for both game loop and server to shut down.
+	for w.IsGameLoopRunning() {
+		time.Sleep(1 * time.Millisecond)
+	}
+	for txh.IsServerRunning() {
```

Comment:
If you remove the `isServerRunning`, this will need to be replaced with a helper that hits the healthcheck endpoint until it no longer returns a 200.

Reply:
I think isServerRunning works fine. If the functionality is the same then I think it's ok to keep it. We can talk about it at standup.

Reply:
As discussed in the standup I removed isServerRunning and i'm doing the healthcheck thingie here.